plugins/kubernetes-backend/src/types/types.ts

```diff
@@ -80,6 +80,17 @@ export type KubernetesObjectTypes =
 
 // Used to load cluster details from different sources
 export interface KubernetesClustersSupplier {
+  /**
+   * Refreshes the list of cluster from the source.
+   *
+   * This will be called periodically on a schedule to refresh the list
+   * of clusters.
+   */
+  refreshClusters(): Promise<void>;
```

Comment:
I find that it's unfortunate that this got hardwired into the interface such that the "framework" (the builder) needs to understand how to refresh things. An alternative approach would have been to keep this interface unchanged, and instead just adding the ability to the implementation class to refresh itself in any way that you see fit. Then you could even have made it able to react on k8s events and be event driven for instant refreshes and stuff, without the rest of the code even knowing about the change. Then you wouldn't need the `setClusterRefreshInterval` change either. You could just do

```ts
builder.setClusterSupplier(
  MyCustomSupplier.create({ refreshIntervalMinutes: 30 })
)
```

or whatever, and everything else just continues working as it used to. Right?

Reply:
Yeah. You could have a `defaultClusterRefreshInterval?: Duration | undefined` field in the options to `createBuilder` if you want (if not given, a hard coded default is used), and the default ones could consume that

Reply:
Okay, so I'll make the existing suppliers refresh themselves using that default and `runPeriodically` internally.  And I'll implement the custom supplier example as you have suggested above.  I'll also use Duration again, that's a much nicer interface than the raw milliseconds.

Reply:
Done and just pushed, I hope this looks good now.